webapp/src/components/AssessmentMetadataBar.tsx

```diff
@@ -0,0 +1,260 @@
+          <>
+            <ListItem>
+              <ListItemAvatar>
+                <Avatar>
+                  <CheckCircleIcon />
+                </Avatar>
+              </ListItemAvatar>
+              <ListItemText secondary="Score" primary={submission.score} />
+            </ListItem>
+          </>
+        )}
+        {submission.state === SubmissionStatus.Opened && (
+          <ListItem>
+            <ListItemAvatar />
+            <ListItemText
+              secondary="Attempts"
+              primary={`${submission.id} out of max ${assessment?.maxNumSubmissions}`}
+            />
+          </ListItem>
+        )}
+        {(submission.state === SubmissionStatus.Submitted ||
+          submission.state === SubmissionStatus.Graded) && (
+          <>
+            <ListItem>
+              <ListItemAvatar>
+                <Avatar>
+                  <InventoryIcon />
+                </Avatar>
+              </ListItemAvatar>
+              <ListItemText
+                secondary="Submissions"
+                primary={`${submission.id} out of ${assessment?.maxNumSubmissions}`}
+              />
+            </ListItem>
+            <ListItem>
+              <ListItemAvatar />
+              <ListItemText
+                secondary="Submitted At"
+                primary={formatDateTime(
+                  new Date(submission.submitAt!).toString()
+                )}
+              />
+            </ListItem>
+          </>
+        )}
+        <Divider variant="middle" component="li" />
+        <ListItem>
+          <ListItemAvatar>
+            <Avatar
+              sx={{
+                bgcolor: `${
+                  submission.state === SubmissionStatus.Opened
+                    ? blue[600]
+                    : grey[400]
+                }`,
+              }}
+            >
+              <CalendarMonthIcon />
+            </Avatar>
+          </ListItemAvatar>
+          <ListItemText
+            secondary="Due Date"
+            primary={formatDateTime(
+              assessment?.dueDate ? assessment?.dueDate : ''
+            )}
+          />
+        </ListItem>
+        {assessment!.type === AssessmentType.Test &&
+          submission.state === SubmissionStatus.Opened && (
+            <>
+              <ListItem>
+                <ListItemAvatar>
+                  <Avatar
+                    sx={{
+                      bgcolor: `${
+                        submission.state === SubmissionStatus.Opened
+                          ? blue[600]
+                          : grey[400]
+                      }`,
+                    }}
+                  >
+                    <TimerIcon />
+                  </Avatar>
+                </ListItemAvatar>
+                <ListItemText
+                  secondary={showTimer ? `Time Remaining` : `Time Limit`}
+                  primary={
+                    showTimer
+                      ? `${Math.floor(secondsRemaining / 60)}m ${
```

Comment:
This should only show minutes if there are minutes remaining; for under a minute the minutes should be hidden.

Perhaps we should add in hours here too?